playbooks/files/rax-maas/plugins/ironic_capacity_check.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python
+
+# Copyright 2017, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import time
+
+import ipaddr
+from ironicclient import exc
+from maas_common import get_auth_ref
+from maas_common import get_ironic_client
+from maas_common import metric
+from maas_common import metric_bool
+from maas_common import print_output
+from maas_common import status_err
+from maas_common import status_ok
+
+
+def check(auth_ref, args):
+
+    IRONIC_ENDPOINT = ('http://{ip}:6385/v1'.format(ip=args.ip))
+
+    try:
+        if args.ip:
+            ironic = get_ironic_client(endpoint=IRONIC_ENDPOINT)
+        else:
+            ironic = get_ironic_client()
+
+        is_up = True
+
+    except exc.ClientException:
+        is_up = False
```

Comment:
Yes,  by looking at:
https://github.com/openstack/python-ironicclient/blob/69b4dda421ec18b8dc61bb0396fd38da7498f746/ironicclient/common/apiclient/exceptions.py#L33

All 4xx, 5xx responses will throw subclasses of Client exception. I think that's a pretty comprehensive list.